R/tt_toString.R

```diff
@@ -590,29 +590,30 @@ setMethod("get_formatted_cells", "ElementaryTable",
 #' @rdname gfc
 setMethod("get_formatted_cells", "TableRow",
           function(obj, shell = FALSE) {
-            default_format <- if (is.null(obj_format(obj))) "xx" else obj_format(obj)
-            format <- lapply(row_cells(obj), function(x) {
-                format <- obj_format(x)
-                if (is.null(format))
-                    default_format
-                else
-                    format
-            })
-            default_na_str <- obj_na_str(obj) %||% "NA"
-            na_str <- vapply(row_cells(obj), function(x) {
-                nastr <- obj_na_str(x)
-                if(is.null(nastr))
-                    default_na_str
-                else
-                    nastr
-            }, "")
-
-            matrix(unlist(Map(function(val, format, spn, na_str) {
+
+            # Parent row format and na_str
+            pr_row_format <- if (is.null(obj_format(obj))) "xx" else obj_format(obj)
+            pr_row_na_str <- obj_na_str(obj) %||% "NA"
+
+            matrix(unlist(Map(function(val, spn) {
                 stopifnot(is(spn, "integer"))
-                val <- if(shell) format else paste(format_rcell(val, format, na_str = na_str),
-                                                   collapse = ", ")
+
+                # Case it is shell
+                if (shell) {
+                    format <- obj_format(val)
```

Comment:
This may be a good argugment for actually pushing `shell` behavior down to the `format_rcell` level. 

That or having `resolve_format(cell, row)` and `resolve_na_str(cell, row)` functions to call here, as I'm not a fan of this logic appearing in 2 places.

Reply:
yes, pushing `shell` down was more elegant, but as you said less useful to have such a variable for `format_rcell`. I will think what is best imo and propose it to you here

Reply:
The current solution looks to me like the best. Still, I had no idea that format can have all these possible values and behaviors. Something showing this schematically is needed imo.